### PR TITLE
Issue-249 Visualize videos OT-496

### PR DESCRIPTION
**Pull request by florianhaar**
**Link to the given issue**
https://github.com/open-xchange/ox-coi/issues/249

**Describe what the problem was / what the new feature is**
Videos were shown as generic attachments and it was not possible to see what a video is.

**Describe your solution**
Added a method to create a thumbnail of the video and show the duration on the image. A play icon shows that the image is a video.

**Additional context**
- Added the video utils class.
- **Before we rebase we need this iOS plugin ticket:**
https://github.com/open-xchange/flutter-deltachat-core/issues/63

### CODE REVIEW

**Comment by Boehrsi** [comment]
Removed the warning level as both Android and iOS implemented the native parts: https://github.com/open-xchange/flutter-deltachat-core/commit/b88f4ebfb0effb457d12c7a72fe00e9dbea0dde3

**Comment by Boehrsi** [comment]
Re-added the _warning_ label as it seems that some file formats are not well handled (no preview is generated). @florianhaar please update this ticket as soon as you have more information. If all issues are fixed please remove the _warning_ label and ping me.

**Comment by Boehrsi** [comment]
As the core is already merged, but this PR needs some adjustments, I added https://github.com/open-xchange/ox-coi/commit/4092eb812162840f349435ed48e14051ed35cab4 as temporary fix. This fix should get removed as soon as this PR is merged.

File: lib/src/message/message_builder.dart [hunk 1]
```diff
@@ -247,6 +313,32 @@ class _MessagePartImageAttachmentState extends State<MessagePartImageAttachment>
     );
   }
 
+  void loadThumbnail() async {
+    imageProvider = MemoryImage(kTransparentImage);
+    String videoPath = _getMessageStateData(context).attachmentStateData.path;
+    String thumbnailPath = "";
+    int dotIndex = videoPath.lastIndexOf('.');
+    String videoPathWithoutEnding = videoPath.substring(0, dotIndex);
+    thumbnailPath = "$videoPathWithoutEnding.jpg";
```
Comment (Boehrsi) [inline lib/src/message/message_builder.dart]: We should probably do a followup ticket removing all hard coded file types / moving those definitions to a helper / constant file.

File: lib/src/message/message_builder.dart [hunk 2]
```diff
@@ -247,6 +313,32 @@ class _MessagePartImageAttachmentState extends State<MessagePartImageAttachment>
     );
   }
 
+  void loadThumbnail() async {
+    imageProvider = MemoryImage(kTransparentImage);
+    String videoPath = _getMessageStateData(context).attachmentStateData.path;
+    String thumbnailPath = "";
+    int dotIndex = videoPath.lastIndexOf('.');
```
Comment (Boehrsi) [inline lib/src/message/message_builder.dart]: Looks like something we should put in a helper or research if there is already a getFileExtension method in Dart iself.

**Comment by Boehrsi** [comment]
No change request, just some comments and potentially stuff for a followup ticket.

File: lib/src/message/message_builder.dart (hunk 1 above)
Comment (florianhaar) [reply lib/src/message/message_builder.dart]: done in https://github.com/open-xchange/ox-coi/pull/298/commits/da49c93d52a3c3b67204f870a88b4cfe2e5123eb

File: lib/src/message/message_builder.dart (hunk 2 above)
Comment (florianhaar) [reply lib/src/message/message_builder.dart]: done in https://github.com/open-xchange/ox-coi/pull/298/commits/da49c93d52a3c3b67204f870a88b4cfe2e5123eb

**Comment by florianhaar** [comment]
Moved the loading logic to the MessageAttachmentBloc

**Pull request merged**